docs/design/architecture-refactor.md

```diff
@@ -0,0 +1,108 @@
+<div style="text-align: center;">single cluster</div>
+
+![Narrows for multi-cluster](https://github.com/4everming/cloud-native-security-inspector/blob/doc/commit-the-design-doc-for-new-architecture-of-Narrows/docs/pictures/architecture-refactor-multi-cluster.png)
+<div style="text-align: center;">multi-cluster</div>
+
+## Detailed Design
+There are several components as the picture showing above which can be divided into 3 categories:   
+### 1. Components on the control plane node:
+* **Manager(Portal):**  
+A component that allows user to set up the configurations and security policies of Project Narrows, and acquire the scanning reports
+* **Resource Collector:**  
+Talk to Kubernetes API server to collect the Kubernetes workloads related info. In addition to that, 
+it also talks to external image registries such as Harbor and DevOps platform such as GitLab to collect the security data.
+* **Database:**  
+It is a relational database to store the data collected from **Resource Collector** and **Data Collector**
+* **Scanner**  
+It is a component to scan the resource based on the data which has already been gathered to databases according to 
+the polices defined by user. After that it will generate the scanning reports. It can be triggered manually or periodically.
+* **Data Exporter**  
+It is a component to export the scanning reports generated by **Scanner** to other places outside Project Narrows so that
+other system can consume the data easily and make capabilities of Project Narrows can be integrated by others.
+* **Rule Engine Connector**  
+It is a component to connect to the policy engine such as OPA(Open Policy Agent).
+### 2. Components on each worker node:
+* **Data Collector**  
+Data Collector will collect all the security related data from Kubernetes worker node and sync the data to database on 
+control plane node, including the security data of worker node and containers on this worker node. 
+* **Action Enforcer**  
+It will receive the action items from components on control plane node.
+### 3. External components:
+* **Image registries**  
+It is the place to manage the image artifacts and scan the images to produce security data as well.
+* **DevOps platforms**  
+It is the place to track the source code and generate the scanning reports based on the code.
+## Roadmap
+### Phase 0
+This will introduce a new runtime security tool "Falco" to Project Narrows with the refactored architecture. In order to make it
+easy and quick for the first deliverables, we will only implement the necessary parts and keep other parts of the 
+project the same as before. So you should expect the Cronjob resource created by Narrows are still existed in Kubernetes
+and some Daemonsets are introduced.
+
+### Phase 1
+In this phase, we plan to remove all the Cronjob resources created by Project Narrows and move all the functionalities to
```

Comment:
I think Kubebench should be changed to daemonset, there is no doubt of that.
But things like image scanner should not be daemonSet, they should be a deployment in the workload cluster, it should be able to call the API server of the workload cluster to fetch all the image information.

So do we really need to eliminate Cronjobs?

Reply:
In the new architecture, all the stuff of scanning will be done by Scanner.  Cronjob is no longer needed.